File: Task.py
class Task:

    def __init__(self, title, id):
        self.title = title
        self.status = "[incomplete]"
        self.id = id
    
    def __init__(self, id, status, title):
        self.id = id
        self.setStatus(status)
        self.title = title

    def getTitle(self):
        return self.title

    def getStatus(self):
        return self.status

    def setStatus(self, status):
        self.status = status

    def getId(self):
        return str(self.id)

    def toString(self):
        return "/"+ self.getId() + "/" + self.getStatus() + "/" + self.getTitle() + "|"
    
    def toText(self):
        return "ID: "+ self.getId() + " " + self.getStatus() + " " + self.getTitle()

File: task.js
class Task {
  constructor(id, status, title, description, location, date, deadline) {
      this.id = id;
      this.status = status;
      this.title = title;
      this.description = description;
      this.location = location;
      this.date = date;
      this.deadline = deadline;
  }

  toText() {
      return {
          id: this.id,
          status: this.status,
          title: this.title,
          description: this.description,
          location: this.location,
          date: this.date,
          deadline: this.deadline
      };
  }

  setStatus(newStatus) {
      this.status = newStatus;
  }

  setID(newId) {
      this.id = newId;
  }
}

module.exports = Task;
